java/src/org/openqa/selenium/json/Json.java

```diff
@@ -27,20 +27,117 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * The <b>Json</b> class is the entrypoint for the JSON processing features of the Selenium API.
+ * <p>
+ * These features include:
+ * <ul>
+ *   <li>Built-in serialization of primitives and collections.</li>
+ *   <li>Facilities to serialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare getter methods that adhere to the
+ *         <a href="https://docs.oracle.com/javase/tutorial/javabeans/writing/index.html">JavaBean</a>
+ *         specification.</li>
+ *       <li>Classes that declare a {@code toJson()} method that returns primitives and collections.</li>
+ *     </ul>
+ *   </li>
+ *   <li>Build-in deserialization of primitives and collections.</li>
+ *   <li>Facilities to deserialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare setter methods that adhere to the {@code JavaBean} specification.</li>
```

Comment:
Probably worth pointing out that using the javabean spec makes instances mutable. The only way to get immutable instances is to use the `fromJson` method

Reply:
I added a note to this effect.